native/cocos/scene/SubModel.cpp

```diff
@@ -218,7 +220,16 @@ void SubModel::onPipelineStateChanged() {
 }
 
 void SubModel::onMacroPatchesStateChanged(const ccstd::vector<IMacroPatch> &patches) {
-    _patches = patches;
+    if (!patches.empty() && !patches.empty()) {
```

Comment:
!patches.empty() && !patches.empty() -> patches.empty() && patches.empty()

Reply:
And i think it is better to change like this
```c++
if (_patches.empty()) _patches = patches;
else if (patches.empty()) _patches.clear();
else {
    // other logic;
}
```

Reply:
And may be doesn't have to check it. If both of `patches` and `_patches` are empty, they can also use the same logic below. As empty equals empty.